website_supplier_list/controllers/main.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+# © 2016 Michael Viriyananda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import SUPERUSER_ID
+from openerp.addons.web import http
+from openerp.addons.website.models.website import unslug
+from openerp.tools.translate import _
+from openerp.addons.web.http import request
+import werkzeug.urls
+
+
+class WebsiteSupplier(http.Controller):
+    _references_per_page = 20
+
+    @http.route([
+        '/suppliers',
+        '/suppliers/page/<int:page>',
+        '/suppliers/country/<int:country_id>',
+        '/suppliers/country/<country_name>-<int:country_id>',
+        '/suppliers/country/<int:country_id>/page/<int:page>',
+        '/suppliers/country/<country_name>-<int:country_id>/page/<int:page>',
```

Comment:
All would be much easier if you use `<model('res.country')>` as the data type anywhere a country is used. You will get the orm object directly as a parameter, and it will redirect ids to slugs and transform slugs in records.